### PR TITLE
added ne/le/ge/expand_as operators

### LINKED ISSUE #1333: Adding not equal operator and expand_as operator.
It seems like ONNX does not have operators for pytorch's expand_as and ne (not equal) operators.

Can I work on adding these operators to ONNX?

### PR DESCRIPTION
**Pull request by ArmenAg**
fixes #1333

### CODE REVIEW

**Comment by houseroad** [comment]
Can you sign the CLA?

**Comment by ArmenAg** [comment]
I have signed it.

**Comment by houseroad** [comment]
It shows you have one more to sign :-)

**Comment by ArmenAg** [comment]
Sorry git troubles :)

**Comment by prasanthpul** [comment]
Agree with @bddppq.

**Comment by prasanthpul** [comment]
This seems like a good use for functions (https://github.com/onnx/onnx/blob/master/docs/Functions.md). Functions are basically compound ops and enable optimization in backends that can but still allow others to run the models without implementing lots of ops.

**Comment by ArmenAg** [comment]
In that case should things like sigmoid\tanh be functions and not operators?

**Comment by snnn** [comment]
No test case?

**Comment by ArmenAg** [comment]
Are the only tests I need to implement [here](https://github.com/onnx/onnx/tree/master/onnx/backend/test/case/node)? What's the consensus on which of these ops need to be added?

**Comment by houseroad** [comment]
You also need to generate the data using these tests. `pythno onnx/backend/test/cmd_tools.py generate-data`

**Comment by ArmenAg** [comment]
I wrote all the tests for ne/ge/le/expand_as.

File: onnx/defs/logical/defs.cc [hunk 1]
```diff
@@ -121,6 +121,48 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+            "Constrains output to boolean tensor."));
+
+ONNX_OPERATOR_SET_SCHEMA(
+    LE,
+    8,
+    OpSchema()
+        .FillUsing(BinaryLogicDocGenerator("less than or equal"))
+        .TypeConstraint(
+            "T",
+            {"tensor(bool)", "tensor(int32)", "tensor(int64)"},
```
Comment (snnn) [inline onnx/defs/logical/defs.cc]: What's the semantic of ‘<=' on bool?

Comment (ArmenAg) [reply onnx/defs/logical/defs.cc]: This is a good point. I wanted to stay consistent with numpy.less_equal functionality which would return a tensor filled with True. But I see the argument as to not supporting boolean tensors for this op.

Comment (ArmenAg) [reply onnx/defs/logical/defs.cc]: What we could do is to maintain support <= on boolean tensors (for consistency reasons), but write an optimization pass that would replace this comparison with a Boolean Tensor of True values.

File: onnx/backend/test/case/node/ge.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from ..base import Base
+from . import expect
+
+
+class GE(Base):
+
+    @staticmethod
+    def export():  # type: () -> None
+        node = onnx.helper.make_node(
+            'LE',
```
Comment (zrphercule) [inline onnx/backend/test/case/node/ge.py]: I think this should be GE?

File: onnx/backend/test/case/node/ge.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        x = (np.random.randn(3, 4, 5) * 10).astype(np.int32)
+        y = (np.random.randn(3, 4, 5) * 10).astype(np.int32)
+        z = np.greater_equal(x, y)
+        expect(node, inputs=[x, y], outputs=[z],
+               name='test_ge')
+
+    @staticmethod
+    def export_equal_broadcast():  # type: () -> None
+        node = onnx.helper.make_node(
+            'NE',
```
Comment (zrphercule) [inline onnx/backend/test/case/node/ge.py]: And here?

File: onnx/backend/test/case/node/le.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        x = (np.random.randn(3, 4, 5) * 10).astype(np.int32)
+        y = (np.random.randn(3, 4, 5) * 10).astype(np.int32)
+        z = np.less_equal(x, y)
+        expect(node, inputs=[x, y], outputs=[z],
+               name='test_le')
+
+    @staticmethod
+    def export_equal_broadcast():  # type: () -> None
+        node = onnx.helper.make_node(
+            'NE',
```
Comment (zrphercule) [inline onnx/backend/test/case/node/le.py]: Should be LE?

**Comment by ArmenAg** [comment]
For consistency with numpy it makes sense. Logical operators should be defined for all numerical types.

**Comment by ArmenAg** [comment]
@bddppq @snnn  @zrphercule What are your opinions on this PR? What are the next steps?

**Comment by spandantiwari** [comment]
May I suggest that we reconsider the names for these three ops, `NE`, `LE`, `GE`. We currently have op names such as `Less`, `Greater`, `Equal`. It would be more consistent naming if we choose something like `NotEqual`, `LessEqual`, `GreaterEqual` for these new ops.

**Pull request closed without merging**